### PR TITLE
Next generation simulator

### PR DESCRIPTION
**Pull request by fakufaku**
Adding much needed feature to the RIR generator

### Done

* Ray tracing for tail simulation
* Frequency dependent absorption
* Air absorption
* Fixed API
* Fixed tests

TODO are tracked in [Project Board 2](https://github.com/LCAV/pyroomacoustics/projects/2)

### Checklist

- [ ] Are there docstrings ? Do they follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) style ?
- [ ] Have you run the tests by doing `nosetests` or `py.test` at the root of the repo ?
- [ ] Have you checked that the doc builds properly and that any new file has been added to the repo ? How to do that is covered in the [documentation](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html#documentation).
- [ ] Is there a unit test for the proposed code modification ? If the PR addresses an issue, the test should make sure the issue is fixed.
- [ ] Last but not least, did you document the proposed change in the CHANGELOG file ? It should go under "Unreleased".

Happy PR :smiley:

### CODE REVIEW

File: pyroomacoustics/libroom_src/room.cpp [hunk 1]
```diff
@@ -0,0 +1,1026 @@
...
+
+  corners << rs[0], 0.f, 0.f, rs[0],
+             0.f, 0.f, rs[1], rs[1],
+             0.f, 0.f, 0.f, 0.f;
+  walls.push_back(Wall<3>(corners, abs.col(4), scat.col(4), "floor"));
+
+  corners << rs[0], rs[0], 0.f, 0.f,
+             0.f, rs[1], rs[1], 0.f,
+             rs[2], rs[2], rs[2], rs[2];
+  walls.push_back(Wall<3>(corners, abs.col(4), scat.col(4), "ceiling"));
```
Comment (ebezzam) [inline pyroomacoustics/libroom_src/room.cpp]: should this be using `abs.col(5)` and `scat.col(5)`? or is it assumed floor and ceiling have same coefficients?

**Comment by fakufaku** [comment]
Nice save! Thanks!

Robin
–––––
GnuPG KeyID: 0x7F0F430C 
Fingerprint: 75FA B2E3 42AD 593C FE81  39A1 B34A C543 7F0F 430C

File: pyroomacoustics/libroom_src/room.cpp (hunk 1 above)
Comment (fakufaku) [reply pyroomacoustics/libroom_src/room.cpp]: Oh! Nice save, thanks! Obviously, it should be `abs.col(5)` and `scat.col(5)`!

**Comment by fakufaku** [comment]
@CyrilCadoux @ebezzam After a very long time, this is finally merged into master!!! I will get this onto pypi ASAP. Thank you so much!!

**Pull request merged**